### ISSUE TITLE
Can't build on Debian Buster

### ISSUE THREAD

**Issue by tanshoku** [trunkmaster/nextspace#305]
I'm using the `Packaging/Debian/build-all.sh` to try and build a Debian version, it seems to create a fake root partition, install a bunch of things in it with apt and so on, then it gets to this part:

```
Install the project...
/usr/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/Block_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/lib/libBlocksRuntime.so
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/base.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/block.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/data.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/dispatch.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/group.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/introspection.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/io.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/object.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/once.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/queue.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/semaphore.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/source.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/time.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_after.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_api.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_apply.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_async.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_data_create.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_group_create.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_io_create.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_io_read.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_object.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_once.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_queue_create.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_read.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_semaphore_create.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_source_create.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/share/man/man3/dispatch_time.3
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/os/object.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/os/generic_unix_base.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/os/generic_win_base.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/benchmark.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/data_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/introspection_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/io_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/layout_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/mach_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/queue_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/source_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/time_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/dispatch/workloop_private.h
-- Installing: /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/lib/libdispatch.so
make[2]: Leaving directory '/build/libdispatch-5.1.5/obj-x86_64-linux-gnu'
rm /build/libdispatch-5.1.5/debian/tmp/usr/NextSpace/include/Block_private.h
make[1]: Leaving directory '/build/libdispatch-5.1.5'
   dh_install
   dh_installdocs
   dh_installchangelogs
   dh_perl
   dh_link
   dh_strip_nondeterminism
   dh_compress
   dh_fixperms
   dh_missing
   dh_dwz
dwz: debian/libdispatch0/usr/lib/debug/.dwz/x86_64-linux-gnu/libdispatch0.debug: .debug_info section not present
dh_dwz: dwz -q -mdebian/libdispatch0/usr/lib/debug/.dwz/x86_64-linux-gnu/libdispatch0.debug -M/usr/lib/debug/.dwz/x86_64-linux-gnu/libdispatch0.debug -- debian/libdispatch0/usr/NextSpace/lib/libBlocksRuntime.so debian/libdispatch0/usr/NextSpace/lib/libdispatch.so returned exit code 1
make: *** [debian/rules:21: binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
I: copying local configuration
E: Failed autobuilding of package
I: unmounting dev/ptmx filesystem
I: unmounting dev/pts filesystem
I: unmounting dev/shm filesystem
I: unmounting proc filesystem
I: unmounting sys filesystem
I: cleaning the build env 
I: removing directory /var/cache/pbuilder/build/4264 and its subdirectories
```

and I can't proceed from here. The catch seems to be `libdispatch0.debug: .debug_info section not present`, but I can't find this file, since, I suppose, it was on `/var` and was deleted, and a quick google search didn't yield anything useful. I also tried to follow the build instructions on `/Packaging/Debian/README.md`, with the only difference being that I ran `sudo pbuild create --distribution buster` instead of `...sid`, but it fails with the same problem. From what I read, this DE/WM  is originally intended to be used with CentOS 7, or at least the pre-built RPMs are, but I'm still really curious to try it out on Debian if possible.

**Comment by zeroheure** [comment]
Current Debian packaging is for Sid but it is not yet ready.
Centos 7 is the reference platform for reasons explained in the README. However, there is packages available for Centos 7, 8 and Fedora 31. 
We are some volunteers working on this, trying to improve current packages and to add new targets. If you want to help...

**Comment by tanshoku** [comment]
Sure, I'm going to try to help if I can figure this out. From what I searched, there was an upgrade on dwz v0.13 that doesn't exit with an error code when a `.debug_info` section is not present, so I have to figure a workaround for that on dwz 0.12. I don't know much about builders, so it might take a while for me to find where this file is or needs to be.   Meanwhile, I tried to build for sid and almost got it but it crashed while building `nextspace-frameworks` due to a checksum mismatch, so I'll try to build it again on a fresh copy of the git repo.

**Comment by zeroheure** [comment]
Debian packaging is made by Patrick Georgi @pgeorgi. He got several PR regarding Debian packaging not yet merged because Sergii Stoian is on vacation. Apply them before building and or ask him for help. 
I am very busy myself trying to create better RPMs that can reuse already packaged GNUStep libs (like pgeorgi already done) and later will be less distro centric. I work on OBS [in a branch](https://build.opensuse.org/project/show/home:zeroheure:branches:home:nick.lindert) and then merge succesful packages in [our main packaging home](https://build.opensuse.org/project/show/home:nick.lindert). ANY help welcome.

**Comment by pgeorgi** [comment]
That's probably due to different versions of dwz. An issue similar to that is discussed on https://sourceware.org/bugzilla/show_bug.cgi?id=24766. I'll see if I can make dwz errors non-fatal

**Comment by tanshoku** [comment]
Yes, that's the one of the pages I looked at about dwz. Also, I managed to build the Debian Sid version, but it took a whole day, and another day to set up a VM and install, and yet I can't start it at all, and if i try to run any of the .app programs, they just crash. If you manage to fix the dwz problem I might try to build it or Debian Buster, but I think it would be ideal if there was a faster way than pbuild to compile and create the .deb packages, as it limits my ability to modify and test things like this.

**Comment by pgeorgi** [comment]
To avoid the crashes, you'll need the nextspace-core package from https://github.com/trunkmaster/nextspace/pull/274 which adds the default config files. Without those, the gnustep libraries search for the wrong GUI backend library.

The build-from-dsc script theoretically allows adding different packaging environments - what would be your preferred one?

**Comment by tanshoku** [comment]
I see. I'll wait for it to be merged or see if i can create the files myself without the package. For packaging, I have used `dpkg-buildpackage`, I think, as a user to build i3-gaps for debian, through the script on [this repo](https://github.com/maestrogerardo/i3-gaps-deb). I wouldn't mind installing things without packaging through `make install` either, but as I tested on CentOS 8, this desktop environment seems to substantially change some of the Unix filesystem entries to a more Apple-ish style (/Users instead of /home, /Applications instead of /usr and /bin, etc), and I'm not sure how that would complicate things if I wanted to switch back to another DE/WM.

**Comment by tanshoku** [comment]
Okay, I see the commit was merged so I'll try to build it again with pbuild still.  @pgeorgi have you figured out how to make dwz exit without an error and not bring down the build? I took some time to understand better how building debian packages work, and I think adding a line like this to `*/debian/rules` would perhaps solve this?

```makefile
override_dh_auto_dwz:
        true
```
or 

```makefile
override_dh_auto_dwz:
        dh_auto_dwz || true
```

is this correct?
